tests/cupy_tests/core_tests/test_ndarray_indexing.py

```diff
@@ -137,6 +140,16 @@ def test_setitem_copy(self, xp, dtype):
         a[:] = b
         return a
 
+    @testing.for_all_dtypes_combination(('src_type', 'dst_type'))
+    @testing.numpy_cupy_array_equal()
+    def test_setitem_different_type(self, xp, src_type, dst_type):
+        a = xp.zeros((2, 3, 4), dtype=dst_type)
+        b = testing.shaped_arange((2, 3, 4), xp, src_type)
+        with warnings.catch_warnings():
+            warnings.filterwarnings('ignore')
```

Comment:
Why don't you specify the warning class?
```suggestion
            warnings.simplefilter('ignore', numpy.ComplexWarning)
```

Reply:
Thanks, I fixed.